.github/workflows/build-and-test-differential.yaml

```diff
@@ -25,6 +25,8 @@ jobs:
   build-and-test-differential:
     needs: make-sure-label-is-present
     if: ${{ needs.make-sure-label-is-present.outputs.result == 'true' }}
+    env:
+      SKIP: ${{ matrix.container-suffix == '-cuda' && !contains(github.event.pull_request.labels.*.name, 'tag:require-cuda-build-and-test') }}
```

Comment:
We had issues around the usage of `github.event.pull_request.labels` in the past that it only gives us the name of the label that was modified. (For example, if you add another tag to the label after `tag:require-cuda-build-and-test`, this variable would be set to false.)

It is better to use the `make-sure-label-is-present` workflow to check if the label is present just like we do for `tag:run-build-and-test-differential` tag.

https://github.com/autowarefoundation/autoware.universe/blob/713d34548742d006b32b4cebc6585b9c1c758370/.github/workflows/build-and-test-differential.yaml#L20

Reply:
I refrect your review in https://github.com/autowarefoundation/autoware.universe/pull/8562 !
Thanks!